en/developer/customization/articles/50-liferay-forms/01-forms-intro.markdown

```diff
@@ -8,18 +8,29 @@ header-id: liferay-forms
 
 The [Liferay Forms](/docs/7-2/user/-/knowledge_base/u/forms) application is
 a full-featured form building tool for collecting data. There's lots of built-in
-functionality, and for the pieces you're missing, there's lots of extensibility.
+functionality. For the pieces you're missing, there are extension points.
 
-This section of tutorials shows developers how to
+This section of articles shows developers how to
 
 1.  Store form entry data in an alternative format. The default storage type is
     JSON.
 
 2.  [Coming Soon] Create new form field types.
 
-This list will continue to grow as new tutorials on customizing and extending
-Liferay Forms are written.
-
-## Customizations and Extension Points
-
-Here's a compilation of the Liferay Forms application's extension points:
+## Liferay Forms Extension Points
+
+Here's a compilation of the Liferay Forms application's extension points that
+are ready for your customization:
+
+- Create a Form Storage Adapter by implementing a 
+    [`StorageAdapter`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/StorageAdapter.java) 
+    or by extending the Abstract implementation,
+    [`BaseStorageAdapter`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/BaseStorageAdapter.java).
+- Create a Form Field Type by implementing a
+    [`DDMFormFieldType`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldType.java),
+    [`DDMFormFieldRenderer`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldRenderer.java),
+    [`DDMFormFieldTypeSettings`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldTypeSettings.java),
+    and a
+    [`DDMFormFieldTemplateContextContributor`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldTemplateContextContributor.java).
+- Create custom validation rules for form fields by implementing a
+    [DDMFormFieldValueValidator](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldValueValidator.java).
```

Comment:
Currently, in regards to the interface `DDMFormFieldValueValidator`, there is a bug that is preventing the error message be displayed to the user, therefore would be necessary we fix it before publishing the doc to the end user. From what I know, there is no LPS ticket for it yet. Is there any Jira ticket associated to this specific extension point? If so, let me know and I can create a LPS ticket registering the mentioned bug and relating to it.

Reply:
It looks like it would be this ticket: https://issues.liferay.com/browse/LPS-95621

Reply:
However, we don't have anything even close to being ready for publication. Here I wanted to mention the extension point and link to the interface, so that third party developers know it is a supported extension point. 

So I think  we should leave it here, fix the bug, then write the docs. Do you agree, @natocesarrego ?

Reply:
Sorry, @rbohl, I meant that there is no LPS registering the bug. In addition, I agree with you that we can leave as it is, fix the bug and then write the doc.